### ISSUE TITLE
minikube start hangs when addon-manager is disabled

### ISSUE THREAD

**Issue by andahme** [kubernetes/minikube#3938]
When addon-manager is disabled, `minikube start` hangs for about 10m before timing out.  After the timeout, the cluster is in a working state as desired.

*  The exact command-lines used so that we can replicate the issue
```
minikube start
```

*  The full output of the command that failed
```
o   minikube v0.35.0 on darwin (amd64)
>   Creating virtualbox VM (CPUs=2, Memory=2048MB, Disk=20000MB) ...
-   "minikube" IP address is 192.168.99.100
-   Configuring Docker as the container runtime ...
-   Preparing Kubernetes environment ...
-   Pulling images required by Kubernetes v1.13.4 ...
-   Launching Kubernetes v1.13.4 using kubeadm ... 
:   Waiting for pods: apiserver proxy etcd scheduler controller addon-manager

*   Sorry that minikube crashed. If this was unexpected, we would love to hear from you:
-   https://github.com/kubernetes/minikube/issues/new
```

*  Which operating system version was used
**macOS 10.4.3**

* Which Minikube version was used
**v0.35.0**

**Comment by tstromberg** [comment]
I agree that we can do better here. Any particular reason for disabling the addon manager?

**Comment by andahme** [comment]
In particular, I have been modifying ingress addon resources (see comments in [2922](https://github.com/kubernetes/minikube/issues/2922)) which have a `Reconcile` mode that makes this tricky.

I did recently discover the `tunnel` feature, which may change how I do ingress in Minikube.  I would make a broader case.

Minikube is my primary k8s experimenting/learning/scratch environment and a test bed for things I plan to take to other environments.  The ability to standup a minimal "vanilla" cluster is pretty awesome.

**Comment by andahme** [comment]
I think the only real issue here is a user experience one.  I downgraded Minikube and reset config (deleted `~/.minikube`) trying to figure out why (it appeared) I could not start a cluster anymore.  In the alternate case (did not hang), I would have realized I needed to enable `addon-manager` again to lay down the base ingress addon resources (and storage-provisioner/etc).

Since `addon-manager` can be disabled and it doesn't appear to be a requirement for cluster operation, I would think it would be ok/good to remove it from the startup wait list.

**Comment by wojciechka** [comment]
As a workaround, what I did was `minikube ssh` and then edit `/etc/kubernetes/manifests/addon-manager.yaml.tmpl`, changing value for  `TEST_ADDON_CHECK_INTERVAL_SEC` to `"60"`.

Then I did `kubectl delete pod -n kube-system kube-addon-manager-minikube`, which caused minikube to restart my pod with a setting of 60 second retrying.

It still consumes CPU for ~3 seconds every minute, but that is something I can now live with and can use `minikube` without noticing huge CPU spike.
